ansible/roles/cli/tasks/deploy.yml

```diff
@@ -30,19 +30,10 @@
     dest: "{{ nginx.confdir }}/cli_temp/{{ openwhisk_cli.archive_name }}.tgz"
   when: openwhisk_cli.installation_mode == "local"
 
-#
-#   I really wanted to use 'unarchive' here, but it was quite buggy and didn't
-#   want to cooperate, so we do a good old-fashioned tar x instead
-#
 - name: "Expand the archive into the build directory"
-  shell: >
-    tar zxf {{ nginx.confdir }}/cli_temp/{{ openwhisk_cli.archive_name }}.tgz
-    -C {{ nginx.confdir }}/cli_temp/{{ openwhisk_cli.archive_name }}/
-
-# Remote copy does not support recursive copy of directories. That's why I'm using the shell.
-- name: "Copy expanded archive to final configuration directory"
-  #  WARNING:  The trailing slash is significant, signalling to copy contents
-  shell: "cp -r {{ nginx.confdir }}/cli_temp/{{ openwhisk_cli.archive_name }}/* {{ openwhisk_cli.nginxdir.name }}"
+  unarchive:
```

Comment:
Due to the documentation of this module, it first tries to copy the archive from the local machine to the remote machine.
If the edge is not the same like the machine where ansible runs, this PR won't work without the flag `remote_src: yes`, as the archive is downloaded to the remote machine with the `get_url` module.